app/src/main/java/org/wordpress/pioupiou/postlist/PostListActivity.java

```diff
@@ -29,37 +46,102 @@
     private SwipeRefreshLayout mSwipeRefreshLayout;
 
     // State
-    boolean mNewPostVisible;
+    private boolean mNewPostVisible;
+    private boolean mIsFetchingPosts;
+
+    private String mNewPostContent;
 
     // FluxC
-    // TODO: Inject stores here
+    @Inject Dispatcher mDispatcher;
+    @Inject PostStore mPostStore;
+    @Inject SiteStore mSiteStore;
+    @Inject AccountStore mAccountStore;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        ((PioupiouApp) getApplication()).component().inject(this);
+
         setContentView(R.layout.activity_post_list);
 
         mSwipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.swipe_to_refresh);
         mSwipeRefreshLayout.setOnRefreshListener(
                 new SwipeRefreshLayout.OnRefreshListener() {
                     @Override
                     public void onRefresh() {
-                        fetchPosts();
+                        if (!mIsFetchingPosts) {
+                            fetchPosts();
+                        }
                     }
                 }
         );
+
+        // immediately show existing posts then fetch the latest from the server
+        showPosts();
+        fetchPosts();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mDispatcher.unregister(this);
+    }
+
+    private void showPosts() {
+        Timber.i("Show posts started");
+        SiteModel site = getSite();
+        if (site == null) {
+            Timber.w("Can't show posts for null site");
+            return;
+        }
+
+        if (hasPostFragment()) {
+            getPostFragment().setPosts(mAccountStore.getAccount(), mPostStore.getPostsForSite(site));
+        }
     }
 
     private void fetchPosts() {
         Timber.i("Fetch posts started");
-        // TODO: fetch posts
-        mSwipeRefreshLayout.setRefreshing(false);
+        SiteModel site = getSite();
+        if (site == null) {
+            Timber.w("Can't fetch posts for null site");
+            return;
+        }
+
+        mIsFetchingPosts = true;
+        mDispatcher.dispatch(PostActionBuilder.newFetchPostsAction(
+                new PostStore.FetchPostsPayload(site)));
+        mSwipeRefreshLayout.setRefreshing(true);
+    }
+
+    /*
+     * returns the site to use for the post list - currently always returns the very first site
+     */
+    private SiteModel getSite() {
+        List<SiteModel> sites = mSiteStore.getSites();
```

Comment:
Yeah, that could be dangerous (especially to us!). Maybe the domain name would be better than the ID, though? We could use `mSiteStore.getSitesByNameOrUrlMatching()` to return the matching site.